app/Services/MetaData.php

```diff
@@ -0,0 +1,157 @@
+    {
+        return (new self($url))->data;
+    }
+
+    /**
+     * Get the meta-data for a given URL.
+     *
+     * @return Collection<string, string>
+     */
+    public function getData(): Collection
+    {
+        $data = collect();
+
+        try {
+            $response = Http::get($this->url);
+
+            if ($response->ok()) {
+                $data = $this->parse($response->body())->filter(fn ($value): bool => $value !== '');
+            }
+        } catch (Exception) {
+            // Do nothing,
+        }
+
+        return $data;
+    }
+
+    /**
+     * Fetch Twitter oEmbed data for a given tweet URL.
+     *
+     * @return Collection<string, string>
+     */
+    private function fetchOEmbed(string $service): Collection
+    {
+        $data = collect();
+
+        try {
+            $response = Http::get(
+                $service.'?url='.urlencode($this->url).'&maxwidth=446&maxheight=251&theme=dark&hide_thread=true&omit_script=true'
+            );
+
+            if ($response->ok()) {
+                $data = collect((array) $response->json())->filter(fn ($value): bool => $value !== '');
+            }
+        } catch (Exception) {
+            // Do nothing,
+        }
+
+        return $data;
+    }
+
+    /**
+     * Parse the response body for MetaData.
+     *
+     * @return Collection<string, string>
+     */
+    private function parse(string $content): Collection
+    {
+        $doc = new DOMDocument();
+        @$doc->loadHTML($content);
+
+        $interested_in = ['og', 'twitter'];
+        $data = collect();
+        $metas = $doc->getElementsByTagName('meta');
+
+        if ($metas->count() > 0) {
+            foreach ($metas as $meta) {
+                // basic meta tags
+                if (mb_strtolower($meta->getAttribute('name')) === 'title') {
+                    $data->put('title', $meta->getAttribute('content'));
+                }
+
+                if (mb_strtolower($meta->getAttribute('name')) === 'description') {
+                    $data->put('description', $meta->getAttribute('content'));
+                }
+                if (mb_strtolower($meta->getAttribute('name')) === 'keywords') {
+                    $data->put('keywords', $meta->getAttribute('content'));
+                }
+
+                // og, twitter cards & other meta tags
+                collect(['name', 'property'])
+                    ->map(fn ($name): string => $meta->getAttribute($name))
+                    ->filter(fn ($attribute): bool => in_array(explode(':', (string) $attribute)[0], $interested_in))
+                    ->each(function ($attribute) use ($data, $meta): void {
+                        $key = explode(':', $attribute)[1];
+                        if (! $data->has($key)) {
+                            $data->put($key, $meta->getAttribute('content'));
+                        }
+                    });
+            }
+        }
+
+        // fetch oEmbed data for X / Twitter
```

Comment:
i think you remove remove docs like this...

Reply:
Done.